### ISSUE TITLE
Non-required values still causing error

### ISSUE THREAD

**Issue by edyu** [GoNZooo/odin-cli#2]
```
args = ["decode", "--output", "blah.txt", "--shard", "encoded", "-N", "5"]
Failed to parse arguments: CliValueParseError{value = "", type = , message = "invalid integer: ''"}
```
The struct is defined below, as you can see, N, K, w are not required.
```
Command :: union {
        Encode,
        Decode,
}

Encode :: struct {
        N:     int `cli:"N,num-code"`,
        K:     int `cli:"K,num-data"`,
        w:     int `cli:"w,word-size"`,
        input: string `cli:"i,input/required"`,
        shard: string `cli:"s,shard/required"`,
}

Decode :: struct {
        N:      int `cli:"N,num-code"`,
        K:      int `cli:"K,num-data"`,
        w:      int `cli:"w,word-size"`,
        output: string `cli:"o,output/required"`,
        shard:  string `cli:"s,shard/required"`,
}
```

**Comment by GoNZooo** [comment]
Thanks for reporting this, Ed. :)

I've fixed the test case you've reported here. Could you try the latest version and see whether it has any other holes that you've seen regarding this?

Overall it should be fixed by #4

**Comment by edyu** [comment]
Fixed from my perspective. :) Thanks.
